src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -238,28 +243,190 @@ private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, ob
             // TCP Format: 
             // tcp:<host name>\<instance name>
             // tcp:<host name>,<TCP/IP port number>
-            int portNumber = 1433;
-            string[] serverAndPortParts = fullServerName.Split(',');
 
-            if (serverAndPortParts.Length == 2)
+            const int DEFAULT_SERVER_PORT = 1433;
+
+            if (String.IsNullOrWhiteSpace(fullServerName))
             {
-                try
+                return new SNITCPHandle("localhost", DEFAULT_SERVER_PORT, timerExpire, callbackObject, parallel);
+            }
+
+            string[] serverNamePartsByComma = fullServerName.Split(',');
+            string[] serverNamePartsByFowardSlash = fullServerName.Split('\\');
```

Comment:
Nit: These are going to allocate `char[]` arrays for the separator because `Split` takes `params char[]`. Instead, the `char[]` arrays can be allocated once and cached in static fields, e.g.:

```c#
private static readonly char[] s_commaSeparator = { ',' };
private static readonly char[] s_backSlashSeparator = { '\\' };

...

string[] serverNamePartsByComma = fullServerName.Split(s_commaSeparator);
string[] serverNamePartsByBackSlash = fullServerName.Split(s_backSlashSeparator);
```